src/Utils/Crypto.php

```diff
@@ -41,11 +42,59 @@ public static function decrypt( $payload, $cipherParams ) {
 
     /**
      * Returns default encryption parameters.
-     * @param $key string|null Encryption key, if not provided a random key is generated.
+     * @param $params Array Array containing optional cipher parameters. A `key` must be specified.
+     * The key may be either a binary string or a base64 encoded string, in which case `'base64Key' => true` must be set.
+     * `iv` can also be provided as binary or base64 string (`'base64IV' => true`), although you shouldn't need it in most cases.
      * @return CipherParams Default encryption parameters.
      */
-    public static function getDefaultParams( $key = null ) {
-        return new CipherParams( $key );
+    public static function getDefaultParams( $params ) {
+        if ( !isset( $params['key'] ) ) throw new AblyException ( 'No key specified.', 40003, 400 );
+
+        $cipherParams = new CipherParams();
+
+        if ( isset( $params['base64Key'] ) && $params['base64Key'] ) {
+            $params['key'] = strtr( $params['key'], '_-', '/+' );
+            $params['key'] = base64_decode( $params['key'] );
+        }
+
+        $cipherParams->key = $params['key'];
+        $cipherParams->keyLength = isset( $params['keyLength'] ) ? $params['keyLength'] : strlen( $cipherParams->key ) * 8;
+        
+        if ( !in_array( $cipherParams->keyLength, array( 128, 256 ) ) ) {
+            throw new AblyException ( 'Unsupported keyLength. Only 128 and 256 bits are supported.', 40003, 400 );
+        }
+        
+        if ( $cipherParams->keyLength / 8 != strlen( $cipherParams->key ) ) {
+            throw new AblyException ( 'keyLength does not match the actual key length.', 40003, 400 );
+        }
+
+        $cipherParams->algorithm = isset( $params['algorithm'] ) ? $params['algorithm'] : 'aes';
+        $cipherParams->mode = isset( $params['mode'] ) ? $params['mode'] : 'cbc';
+
+        if ( !in_array( $cipherParams->getAlgorithmString(), array( 'aes-128-cbc', 'aes-256-cbc' ) ) ) {
+            throw new AblyException ( 'Unsupported cipher configuration "' . $cipherParams->getAlgorithmString()
+                . '". The supported configurations are aes-128-cbc and aes-256-cbc', 40003, 400 );
```

Comment:
Are we sure other algorithms aren't supported if they specified it?  We've taken the view in other libs that we don't block it allowing the user to configure the cipher at their peril